mission_manager/include/mission_manager/behavior.h

```diff
@@ -0,0 +1,416 @@
+/*********************************************************************
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2020, QinetiQ, Inc.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of QinetiQ nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+// Original version: Christopher Scianna Christopher.Scianna@us.QinetiQ.com
+
+#ifndef MISSION_MANAGER_BEHAVIOR_H
+#define MISSION_MANAGER_BEHAVIOR_H
+
+#include <ros/ros.h>
+#include <ros/time.h>
+#include <string.h>
+
+#include <boost/date_time/gregorian/gregorian.hpp>
+#include <boost/date_time/posix_time/posix_time.hpp>
+#include <boost/thread.hpp>
+#include <list>
+#include <map>
+#include <string>
+
+#include "pose_estimator/CorrectedData.h"
+#include "std_msgs/Header.h"
+#include "tinyxml/tinyxml.h"
+
+typedef enum
+{
+  BEHAVIOR_TYPE_MSG = 0,
+  BEHAVIOR_TYPE_SRV
+}
+behavior_type_t;
+
+class TimeStamp
+{
+ public:
+  TimeStamp() {}
+  virtual ~TimeStamp() {}
+
+  typedef enum
+  {
+    REL_TIME = 0,
+    ABS_TIME,
+    ABS_DATETIME
+  }
+  ts_type_t;
+
+  ts_type_t type;
+  boost::posix_time::time_duration duration;
+  boost::posix_time::ptime time;
+};
+
+// class IFactory;
+
+class BehaviorXMLParam
```

Comment:
@patapufate why removing `namespace` scoping? Declaring anything in the global namespace is almost always a bad idea.

Reply:
Restored